cloudformation/infrastructure/fargate-cluster.yaml

```diff
@@ -1,392 +1,407 @@
+    Default: latest
+
+  DjangoKeyId:
+    Type: String
+    Description: unique ID appended to end of DjangoSecretKey ARN in secrets manager
+    Default: xxxxx
+
+  DbSecretId:
+    Type: String
+    Description: unique ID appended to end of DB password ARN in secrets manager
+    Default: xxxxx
+
+  EnvName:
+    Type: String
+    Description: which environment to target
+    AllowedValues:
+      - "dev"
+      - "test"
+      - "stage"
+      - "prod"
+    ConstraintDescription: Must match a location for secret storage in secretsmanager
+
+  FullEnvironmentName:
+    Type: String
+    Description: Full name of deployment environment
+    AllowedValues:
+      - "development"
+      - "test"
+      - "staging"
+      - "production"
+
+  RedisAddress:
+    Type: String
+    Description: Redis endpoint address
+
+  RedisPort:
+    Type: String
+    Description: Redis endpoint port
+
+  MemcachedAddress:
+    Type: String
+    Description: memcached endpoint address
+
+  MemcachedPort:
+    Type: String
+    Description: memcached endpoint port
+
+  CanonicalHostName:
+    Type: String
+    Description: canonical host name of the application, e.g. crowd-test.loc.gov
+
+  DatabaseEndpoint:
+    Type: String
+    Description: Host name of the Postgres RDS service
+
+  S3BucketName:
+    Type: String
+    Description: name of the S3 bucket (public) where collection images will be stored
+
+  ExportS3BucketName:
+    Type: String
+    Description: name of the S3 bucket (public) where exported transcriptions will be stored
 
 Resources:
-    ConcordiaServiceSecretAccessPolicy:
-        Type: AWS::IAM::Policy
-        Properties:
-            PolicyName: !Sub ReadOnlyAccessToConcordiaSecrets-${EnvironmentName}
-            Roles:
-                - !Ref 'ConcordiaTaskRole'
-                - !Ref 'ConcordiaEC2Role'
-            PolicyDocument:
-                Version: '2012-10-17'
-                Statement:
-                    - Effect: Allow
-                      Action:
-                          - 'secretsmanager:GetResourcePolicy'
-                          - 'secretsmanager:GetSecretValue'
-                          - 'secretsmanager:DescribeSecret'
-                          - 'secretsmanager:ListSecretVersionIds'
-                          - 'kms:GetParametersForImport'
-                          - 'kms:GetKeyRotationStatus'
-                          - 'kms:GetKeyPolicy'
-                          - 'kms:DescribeKey'
-                          - 'kms:ListResourceTags'
-                          - 'kms:Decrypt'
-                          - 'kms:GenerateDataKey'
-                          - 's3:PutObject'
-                          - 's3:GetObject'
-                          - 's3:AbortMultipartUpload'
-                          - 's3:ListMultipartUploadParts'
-                          - 's3:ListBucket'
-                          - 's3:ListBucketMultipartUploads'
-                      Resource:
-                          - '*'
```

Comment:
Since this is getting tightened — which is great! — why not break it into multiple statements? The S3 rules seem to be otherwise unrelated and that would allow the linter suppression to only apply to that particular block — especially since it would already need to have both `- !Sub "arn:aws:s3:::crowd-${EnvironmentName}-content/*"` and an additional `- !Sub "arn:aws:s3:::crowd-${EnvironmentName}-content/"` to have `s3:ListBucket` work.

Reply:
Good idea. I'll do that. I actually already did it that in one of the environments because it makes testing easier. The AWS IAM user interface leaves much to be desired.